example/provider-extensions/garden/configure-admission.sh

```diff
@@ -33,12 +33,26 @@ shift 2
 for p in $(yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == "provider-*" or .metadata.name == "networking-*") | .metadata.name' "$REPO_ROOT_DIR"/example/provider-extensions/garden/controllerregistrations/* | grep -v -E "^---$"); do
   echo "Found \"$p\" in $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations. Trying to configure its admission controller..."
   if PROVIDER_RELEASES=$(curl --fail -s -L -H "Accept: application/vnd.github+json" "https://api.github.com/repos/gardener/gardener-extension-$p/releases"); then
-    LATEST_RELEASE=$(jq -r '.[].tag_name' <<< "$PROVIDER_RELEASES" | head -n 1)
     ADMISSION_NAME=$(sed -E 's/(provider|networking)/admission/g' <<< $p)
-    echo "Identified $LATEST_RELEASE as latest release of $ADMISSION_NAME. Trying to $command it..."
+    RELEASE=$(jq -r '.[].tag_name' <<< "$PROVIDER_RELEASES" | head -n 1)
+    echo "Identified \"$RELEASE\" as latest release of $ADMISSION_NAME."
+    IMAGE=$(yq ". | select(.kind == \"ControllerDeployment\") | select(.metadata.name == \"$p\") | .helm.values.image" "$REPO_ROOT_DIR"/example/provider-extensions/garden/controllerregistrations/*)
+    if [ -n "$(yq '.tag' <<< "$IMAGE")" ]; then
+      CONTROLLER_DEPLOYMENT_RELEASE=$(yq '.tag' <<< "$IMAGE")
+    else
+      CONTROLLER_DEPLOYMENT_RELEASE=${IMAGE#*:}
+    fi
+    echo "Identified \"$CONTROLLER_DEPLOYMENT_RELEASE\" as currently used release of \"$p\" ControllerDeployment from its '.providerConfig.values.image.tag' field."
+    if [[ $(jq -r "any(.[].tag_name; . == \"$CONTROLLER_DEPLOYMENT_RELEASE\")" <<< "$PROVIDER_RELEASES") == "true" ]]; then
+      RELEASE=$CONTROLLER_DEPLOYMENT_RELEASE
+    else
+      echo "Could not find \"$CONTROLLER_DEPLOYMENT_RELEASE\" release in the set of github repository releases of \"$p\"."
+      echo "Using latest release \"$RELEASE\" instead."
+    fi
+    echo "Trying to deploy $ADMISSION_NAME with $RELEASE..."
```

Comment:
See https://github.com/gardener/gardener/pull/10472/commits/9e3e85af54a2ae08b786d5f025eade30bfc3f6c6.

:) 

```suggestion
    echo "Trying to $command $ADMISSION_NAME with $RELEASE..."
```